### PR TITLE
gh-105375: Improve error handling in the sys extension module

### PR DESCRIPTION
**Pull request by erlend-aasland**
Bail on first error to prevent exceptions from possibly being
overwritten.

* Issue: gh-105375

### CODE REVIEW

File: Python/sysmodule.c [hunk 1]
```diff
@@ -3374,14 +3380,20 @@ sys_add_xoption(PyObject *opts, const wchar_t *s)
     const wchar_t *name_end = wcschr(s, L'=');
     if (!name_end) {
         name = PyUnicode_FromWideChar(s, -1);
+        if (name == NULL) {
+            goto error;
```
Comment (serhiy-storchaka) [inline Python/sysmodule.c:3384]: `value` is not initialized here.

File: Python/sysmodule.c [hunk 2]
```diff
@@ -3374,14 +3380,20 @@ sys_add_xoption(PyObject *opts, const wchar_t *s)
...
+        }
         value = Py_NewRef(Py_True);
     }
     else {
         name = PyUnicode_FromWideChar(s, name_end - s);
+        if (name == NULL) {
+            goto error;
```
Comment (serhiy-storchaka) [inline Python/sysmodule.c:3391]: Same as above.

File: Python/sysmodule.c (hunk 1 above)
Comment (erlend-aasland) [reply Python/sysmodule.c:3384]: Good catch!

Comment (sobolevn) [reply Python/sysmodule.c:3384]: yes, I have a fix for that https://github.com/python/cpython/issues/105673

Comment (erlend-aasland) [reply Python/sysmodule.c:3384]: Thanks, @sobolevn!

Comment (erlend-aasland) [reply Python/sysmodule.c:3384]: Marking as resolved, as a PR with the fix is expected.

**Pull request merged**